mathesar/tests/imports/test_excel.py

```diff
@@ -0,0 +1,86 @@
+import pytest
+
+from django.core.files import File
+from sqlalchemy import text
+
+from mathesar.models.base import DataFile, Schema
+from mathesar.imports.base import create_table_from_data_file
+from db.schemas.operations.create import create_schema
+from db.schemas.utils import get_schema_oid_from_name
+from db.tables.operations.create import DuplicateTable
+
+TEST_SCHEMA = "import_excel_schema"
+
+
+@pytest.fixture
+def data_file(patents_excel_filepath):
+    with open(patents_excel_filepath, "rb") as excel_file:
+        data_file = DataFile.objects.create(file=File(excel_file), type='excel')
+    return data_file
+
+
+@pytest.fixture()
+def schema(engine, test_db_model):
+    create_schema(TEST_SCHEMA, engine)
+    schema_oid = get_schema_oid_from_name(TEST_SCHEMA, engine)
+    yield Schema.current_objects.create(oid=schema_oid, database=test_db_model)
+    with engine.begin() as conn:
+        conn.execute(text(f'DROP SCHEMA "{TEST_SCHEMA}" CASCADE;'))
```

Comment:
This `schema` fixture isn't needed. I'm aware our fixture setup is pretty esoteric.

Reply:
We need the `schema` fixture because we want to test that we are not able to create duplicate tables with the same name in a schema. 
I tried removing the fixture completely and then tried to create duplicate tables but could not replicate the scenario. If you could provide some more insights on that, I will update it accordingly.

Reply:
No you don't, just get the oid of the schema from the name using `get_schema_oid_from_name` and then trigger reflection in the `Schema` model using `objects` instead of `current_objects`

Reply:
Cool. Updated it now. PTAL